MiniApp/Classes/core/Protocols/MiniAppMessageDelegate.swift

```diff
@@ -150,3 +150,54 @@ public enum MASDKCustomPermissionError: String, MiniAppErrorProtocol {
         }
     }
 }
+
+/// Enumeration that is used to return  Access Token error
+public enum MASDKAccessTokenError: Error, MiniAppErrorProtocol {
+
+    /// Host app failed to implement required interface
+    case failedToConformToProtocol
+
+    /// Requested Audience is not supported
+    case audienceNotSupportedError
+
+    /// Requested Scope is not supported
+    case scopesNotSupportedError
+
+    /// Authorization failed and the reason will be shared by the host app
+    case authorizationFailureError(description: String)
+
+    /// Unknown/Custom error
+    case error(description: String)
+
+    /// Detailed Description for every MASDKAccessTokenError
+    public var description: String {
+        switch self {
+        case .failedToConformToProtocol:
+            return MASDKLocale.localize(.failedToConformToProtocol)
+        case .audienceNotSupportedError:
+            return "The value passed for 'audience' is not supported."
+        case .scopesNotSupportedError:
+            return "The value passed for 'scopes' is not supported."
+        case .authorizationFailureError(let description):
+            return description
+        case .error(let description):
+            return description
+        }
+    }
+
+    /// Title of the error
+    public var name: String {
```

Comment:
maybe we could rename this one `rawValue` to mimic a string enum behavior?

Reply:
No, these error is conformed to `MiniAppErrorProtocol` which has `name` & `description`. We can update them to `rawValue` but that involves in updating the `MiniAppErrorProtocol` itself. Is that okay ?

Reply:
no that's fine, thanks for the heads up ;)